### PR TITLE
Added Semaphore implementation for Windows.

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

File: synchronization/semaphore.cc [hunk 1]
```diff
@@ -48,8 +48,48 @@ class PlatformSemaphore {
...
+  }
+
+  bool IsValid() const { return _sem != nullptr; }
+
+  bool TryWait() {
+    if (_sem == nullptr) {
+      return false;
+    }
+
+    return (WaitForSingleObject(_sem, 0) == 0);
```
Comment (cbracken) [inline synchronization/semaphore.cc]: nit: drop outer parens.

File: synchronization/semaphore.cc [hunk 2]
```diff
@@ -48,8 +48,48 @@ class PlatformSemaphore {
...
-#else  // OS_MACOSX
+#elif OS_WIN
+#include <windows.h>
+
+namespace flutter {
+
+class PlatformSemaphore {
+ public:
+  explicit PlatformSemaphore(uint32_t count)
+      : _sem(CreateSemaphore(NULL, count, 1024, NULL)) {}
```
Comment (cbracken) [inline synchronization/semaphore.cc]: Out of curiosity, why a max count of 1024?

File: synchronization/semaphore.cc (hunk 1 above)
Comment (cbracken) [inline synchronization/semaphore.cc]: Given that the API docs specify a set of `#define`d return values, consider:
```
return WaitForSingleObject(_sem, 0) == WAIT_OBJECT_0;
```

File: synchronization/semaphore.cc (hunk 2 above)
Comment (bkonyi) [reply synchronization/semaphore.cc]: That's a good question... I'd like to ask July 2017 Ben the same thing. Either way, we need to provide a maximum count and there's no limit (and 1024 seems to work for now...).

File: synchronization/semaphore.cc (hunk 1 above)
Comment (bkonyi) [reply synchronization/semaphore.cc]: Will do before merging.

Comment (bkonyi) [reply synchronization/semaphore.cc]: Ah, good catch! Will update.

File: synchronization/semaphore.cc (hunk 2 above)
Comment (cbracken) [reply synchronization/semaphore.cc]: Looks like it's a long -- maybe just `LONG_MAX`?

Comment (bkonyi) [reply synchronization/semaphore.cc]: Makes sense. Done!

**Comment by cbracken** [comment]
lgtm

**Pull request merged**